smartapps/curb/curb-energy-monitor.src/curb-energy-monitor.groovy

```diff
@@ -0,0 +1,477 @@
+        headers: ["Authorization": "Bearer ${state.authToken}"],
+        requestContentType: 'application/json'
+    ]
+
+    asynchttp_v1.get(processUsage, latestparams)
+
+}
+
+def processUsage(resp, data) {
+    if (resp.hasError()) {
+        refreshAuthToken()
+        log.error "Usage Response Error: ${resp.getErrorMessage()}"
+        return
+    }
+    def json = resp.json
+    def main = 0.0
+    def production = 0.0
+    def consumption = 0.0
+    if (json) {
+        json.each {
+            if (!it.main && !it.production) {
+              updateChildDevice("${it.id}", it.label, it.avg)
+              consumption += it.avg
+            }
+            if (it.main) {
+              main += it.avg
+              state.hasMains = true
+            }
+            if (it.production) {
+              if (it.production == "Line-side") {
+                main += it.avg
+              }
+              production += it.avg
+              state.hasProduction = true
+            }
+        }
+        if (state.hasMains && main != 0.0) {
+            updateChildDevice("__NET__", "Total Power Grid Impact", main)
+        }
+        else {
+            updateChildDevice("__NET__", "Total Power Grid Impact", consumption + production)
+        }
+        if (state.hasProduction) {
+            updateChildDevice("__PRODUCTION__", "Total Power Production", production)
+            updateChildDevice("__CONSUMPTION__", "Total Power Usage", consumption)
+        }
+    }
+}
+
+def processDevices(resp, data) {
+    if (resp.hasError()) {
+        log.error "Error setting up devices: ${resp.getErrorMessage()}"
+        return
+    }
+    def json = resp.json
+    state.hasMains = false
+    state.hasProduction = false
+    if (json) {
+        json.circuits.each {
+            if (!it.main && !it.production) {
+                device = createChildDevice("${it.id}", "${it.label}")
+            }
+            if (it.production) {
+                state.hasProduction = true
+            }
+            if (it.main) {
+              state.hasMains = true
+            }
+        }
+
+        if (state.hasProduction) {
+            createChildDevice("__NET__", "Total Power Grid Impact")
+            createChildDevice("__PRODUCTION__", "Total Power Production")
+            createChildDevice("__CONSUMPTION__", "Total Power Usage")
+        } else {
+            //If they don't have production, just call NET Total Power Usage
+            createChildDevice("__NET__", "Total Power Usage")
+        }
+    }
+}
+
+def processKwh(resp, data) {
+    if (resp.hasError()) {
+        refreshAuthToken()
+        log.error "Usage Response Error: ${resp.getErrorMessage()}"
+        return
+    }
+    def json = resp.json
+    def main = 0.0
+    def production = 0.0
+    def existingDevice = null
+    if (json) {
+        json.each {
+            if (!it.main && !it.production) {
+                getChildDevice("${it.id}").handleKwhBilling(it.kwhr)
```

Comment:
You might want to call getChildDevice("${it.id}").handleKwhBilling(it.avg) here, that way only one callback needs to be run and you update both kWh and Power Usage at the same time.

Reply:
Additionally, please use a null check by adding a question mark:

`getChildDevice("${it.id}")?.handlePower(it.avg)`